File: SortingAlgorithms/SortingAlgorithms/BubbleSort.cpp
#include "BubbleSort.hpp"
#include <algorithm>

void bubbleSort(int arr[], int size) {
	bool swapped;
	do {
		swapped = false;
		for (int i = 1; i < size - 1; i++) {
			if (arr[i - 1] > arr[i]) {
				std::swap(arr[i - 1], arr[i]);
				swapped = true;
			}
		}
	} while (swapped);
}

File: SortingAlgorithmsPython/bubbleSort.py
def bubble_sort(arr, swapped = False):
    swapped = False
    for i in range(1, len(arr)):
        if arr[i - 1] > arr[i]:
            arr[i - 1], arr[i] = arr[i], arr[i - 1]
            swapped = True
    if swapped: bubble_sort(arr, swapped)
    else: return arr
    